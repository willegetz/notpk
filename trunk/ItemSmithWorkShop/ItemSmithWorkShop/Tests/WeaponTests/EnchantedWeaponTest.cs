using ItemSmithWorkShop.AdventureItems.MagicWeaponUtilities;
using ItemSmithWorkShop.AdventureItems.MundaneWeaponUtilites;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;

using ItemSmithWorkShop.AdventureItems.ExtraordinaryWeaponUtilities;
using ItemSmithWorkShop.WeaponUtilities;

namespace ItemSmithWorkShop.Tests.WeaponTests
{
	[TestClass]
	public class EnchantedWeaponTest
	{
		[TestMethod]
		public void TestFlamingEnchantment()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(weapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingAdamantineWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon adamantineWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Adamantine");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(adamantineWeapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingSilverWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon silverWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Silver");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(silverWeapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingColdIreonWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon ironWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Cold Iron");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(ironWeapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingDarkwoodItem()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon darkwoodWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Darkwood");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(darkwoodWeapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingMithralItem()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			ExtraordinaryQualityWeapon mithralWeapon = WeaponItemSmith.OrderSpecialComponent(weapon, "Mithral");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(mithralWeapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestWeaponEnchanter()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(weapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestDoubleWeaponEnchanter()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(weapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			EnchantedWeaponItem doubleEnchantedWeapon = WeaponEnchanter.RequestEnchantment(enchantedWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(doubleEnchantedWeapon));
		}

		[TestMethod]
		public void TestFlamingIcyBurstDagger()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(weapon, 1);
			EnchantedWeaponItem flamingWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			EnchantedWeaponItem icyBurstWeapon = WeaponEnchanter.RequestEnchantment(flamingWeapon, "Icy Burst");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(icyBurstWeapon));
		}

		[TestMethod]
		public void TestVorpaldagger()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(weapon, 1);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Vorpal");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}
	}
}
