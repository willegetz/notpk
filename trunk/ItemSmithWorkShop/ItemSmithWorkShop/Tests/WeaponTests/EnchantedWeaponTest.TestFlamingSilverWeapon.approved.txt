+1 Alchemical Silver Dagger, Flaming:	'8322 gp'	('+2 value')
To Hit: '+1'
Damage: '1d4 +0 +1D6 [19-20 x2]' Piercing or Slashing, Magic, Fire
Hardness: '14'
Hit Points: '22'
Weight: '1 pound(s)'
	Alchemical Silver bypasses damage reduction on lycanthropes
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.

	Upon command, a flaming weapon is sheathed in fire.
	The fire does not harm the wielder. The effect
	remains until another command is given.

Creator Caster Level: '10'
Time to Create: '8 Days'
Creation XP Cost: '320'
Creation Raw Material Cost: '4322'

	Required Feats: Craft Magic Arms and Armor
	Required Spells: flame blade, flame strike, or fireball
